src/libDirectoryService/FinalBlockPreProcessing.cpp

```diff
@@ -86,10 +88,10 @@ void DirectoryService::ExtractDataFromMicroblocks(
         if (!isVacuousEpoch && !isEmpty)
         {
             m_mediator.m_node->m_unavailableMicroBlocks[blockNum].insert(
-                {{microBlock.GetHeader().GetTxRootHash(),
-                  microBlock.GetHeader().GetStateDeltaHash()},
+                {{{microBlock.GetHeader().GetTxRootHash(),
+                   microBlock.GetHeader().GetStateDeltaHash()}},
                  // {!isEmptyTxn, true}});
-                 {false, true}});
+                 vector<bool>{false, true}});
```

Comment:
The casting might be redundant because there is no casting in line 981. It's better to make the code be consistent.